sources/packages/backend/apps/api/src/services/education-program/education-program.service.ts

```diff
@@ -271,121 +273,135 @@ export class EducationProgramService extends RecordDataModelService<EducationPro
    * Gets all the programs that are associated with an institution
    * alongside with the total of offerings on locations.
    * @param institutionId id of the institution.
-   * @param paginationOptions pagination options.
-   * @param locationId optional location id to filter.
-   * @returns paginated summary for the institution or location.
+   * @param offeringTypes offering types.
+   * @param paginationOptions program related pagination options.
+   * @returns paginated summary for the institution.
    */
   async getProgramsSummary(
     institutionId: number,
     offeringTypes: OfferingTypes[],
-    paginationOptions: PaginationOptions,
-    locationId?: number,
+    paginationOptions: ProgramPaginationOptions,
   ): Promise<PaginatedResults<EducationProgramsSummary>> {
-    const paginatedProgramQuery = this.repo
-      .createQueryBuilder("programs")
-      .select("programs.id", "programId")
-      .addSelect("programs.name", "programName")
-      .addSelect("programs.cipCode", "cipCode")
-      .addSelect("programs.credentialType", "credentialType")
-      .addSelect("programs.createdAt", "programSubmittedAt")
-      .addSelect("location.id", "locationId")
-      .addSelect("location.name", "locationName")
-      .addSelect("programs.programStatus", "programStatus")
-      .addSelect("programs.isActive", "isActive")
-      .addSelect("programs.effectiveEndDate", "effectiveEndDate")
-      .addSelect(
-        (qb) =>
+    const { programQuery, queryParams } = this.getProgramsQueryWithQueryParams(
+      offeringTypes,
+      institutionId,
+    );
+
+    // Program name search.
+    if (paginationOptions.programNameSearch) {
+      programQuery.andWhere("programs.name Ilike :programNameSearchCriteria", {
+        programNameSearchCriteria: `%${paginationOptions.programNameSearch}%`,
+      });
+      queryParams.push(`%${paginationOptions.programNameSearch}%`);
+    }
+    // Location name search.
+    if (paginationOptions.locationNameSearch) {
+      programQuery.andWhere("location.name Ilike :locationNameSearchCriteria", {
+        locationNameSearchCriteria: `%${paginationOptions.locationNameSearch}%`,
+      });
+      queryParams.push(`%${paginationOptions.locationNameSearch}%`);
+    }
+    // When both the status search and inactive search is false, nothing is returned.
+    if (
+      !paginationOptions.statusSearch &&
+      !paginationOptions.inactiveProgramSearch
+    ) {
+      return {
+        results: [],
+        count: 0,
+      };
+    }
```

Comment:
I am not following up on why "nothing" should be returned for this case. Can you please elaborate?

Reply:
As we talked let's move this to the beginning of the method.
It seems okay to stay if it respects the business requirements. 
My advise would be to UI and DTO enforce that one of these should be present.